tests/unit/MDMMotionCurveTests.m

```diff
@@ -53,4 +53,43 @@ - (void)testSpringCurveDataWithMacro {
   XCTAssertEqualWithAccuracy(curve.data[MDMSpringMotionCurveDataIndexFriction], 0.3, 0.001);
 }
 
+- (void)testSystemModalMovementTimingCurveMatchesModalMovementTiming {
+  UIWindow *window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
+  window.rootViewController = [[UIViewController alloc] init];
+  [window makeKeyAndVisible];
+
+  UIViewController *presentedViewController = [[UIViewController alloc] init];
+  XCTestExpectation *didComplete =
+      [[XCTestExpectation alloc] initWithDescription:@"Animation completed."];
+  [window.rootViewController presentViewController:presentedViewController
+                                          animated:YES
+                                        completion:^{
+                                          [didComplete fulfill];
+                                        }];
+  XCTestExpectation *didExtractAnimationValues =
+      [[XCTestExpectation alloc] initWithDescription:@"Extracted animation values."];
+  dispatch_async(dispatch_get_main_queue(), ^{
+    CAAnimation *animation = [presentedViewController.view.layer animationForKey:@"position"];
```

Comment:
I may be wrong, but this seems like a risky race condition. The "presentation" animation will run in the next loop of the main queue, and your `dispatch_async` will run in the subsequent loop.  Is there no way to use mocking/swizzling to capture the values passed to the presentedViewController instead?

Reply:
Fixed.